core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -899,21 +900,30 @@ private RelationPlan createMergePlan(Analysis analysis, Merge node)
     private PlanNode createOutputPlan(RelationPlan plan, Analysis analysis)
     {
         ImmutableList.Builder<Symbol> outputs = ImmutableList.builder();
-        ImmutableList.Builder<String> names = ImmutableList.builder();
+        ImmutableList.Builder<ColumnInfo> columns = ImmutableList.builder();
 
         int columnNumber = 0;
         RelationType outputDescriptor = analysis.getOutputDescriptor();
         for (Field field : outputDescriptor.getVisibleFields()) {
             String name = field.getName().orElse("_col" + columnNumber);
-            names.add(name);
+            String label = field.getOriginColumnName().orElse(name);
```

Comment:
remove `.orElse(name)`, keep label optional here.
i am aware that somewhere else we also do this logic (`label.or(name)`), and especially because of that we should do same logic here.